docker-images/kafka/scripts/oauth.sh

```diff
@@ -0,0 +1,133 @@
+#!/bin/bash
```

Comment:
What is the purpose of this script? Should it really be added?

Reply:
This is the script that allows you to login as a client or as a user to an OAuth2 authorization server and get an access token or a refresh token which you can then use to authenticate when running kafka CLI tools.

Reply:
Again ... not sure this should be in the container.

Reply:
It is certainly a matter of convenience to have it in this image.
 
It allows for the following usage pattern:
```
kubectl exec -ti my-cluster-kafka-0 -- /bin/sh

ACCESS_TOKEN=$(./oauth.sh --endpoint http://keycloak:8080/auth/realms/kafka-authz/protocol/openid-connect/token --client-id team-a-client --secret team-a-client-secret --access -q)
echo $ACCESS_TOKEN

cat > /tmp/team-a-client-plain.properties << EOF
security.protocol=SASL_PLAINTEXT
sasl.mechanism=PLAIN
sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required \
 username="team-a-client" \
 password="$ACCESS_TOKEN" ;
EOF

bin/kafka-console-producer.sh --broker-list my-cluster-kafka-0:9092 --topic a_messages \
 --producer.config=/tmp/team-a-client-plain.properties
```

Without the `oauth.sh` script present I have two options:
- Use `kubectl cp` to copy the script into the running container, every OAuth example using this pattern would have to include this step
- Use some other image with `oauth.sh` prepackaged. Probably that image won't contain kafka CLI tools, therefore the obtained access token will have to be copied over into the other shell. But in any case, is there another image in the project that would be appropriate for this?

Reply:
Similar to the script above:
* You do not want to run this inside the broker pod
* In real production environments you should not have access to run this inside the pod. So what are the scenarios where you expect to run it?

If any, I think the `test-client` image would be more appropriate for this. Since this seems to be pretty much them same - a tool useful for testing. But it looks like some standard tool for development and testing. So having it in the tools directory seems appropriate.

Reply:
See [above](https://github.com/strimzi/strimzi-kafka-operator/pull/3961#discussion_r524561464).
I'm fine with having it in another image. It want to use it in examples so it should be something we propose people use / consider a 'good practice'.